crates/bevy_reflect/bevy_reflect_derive/src/derive_data.rs

```diff
@@ -411,3 +473,185 @@ impl<'a> ReflectEnum<'a> {
         &self.variants
     }
 }
+
+/// Represents a path to a type.
+pub(crate) enum PathToType<'a> {
+    /// Types without a crate/module that can be named from any scope (e.g. `bool`).
+    Primitive(&'a Ident),
+    /// Using `::my_crate::foo::Bar` syntax.
+    ///
+    /// May have a seperate alias path used for the `TypePath` implementation.
+    External { path: &'a Path, alias: Option<Path> },
```

Comment:
Nit: Type aliases (#5830) may or may not be merged in at some point in the future. _If_ it is, using `alias` here may be confusing. Perhaps we can use something more like `custom_path`? (And `is_aliased` can become `is_custom`.)